### ISSUE TITLE
Ball tracking related questions

### ISSUE THREAD

**Issue by MinseokKang003** [jac99/FootAndBall#2]
Hi, I'm a student interested in sports and deep learning. I came across to your project after watching the Pydata conference, which I found very interesting. So I tested out the code, and it worked perfectly on the ISSIA dataset. But since I'm quite new to deep learning, I have some questions, which I hope to get answers from you.

1) Could there be any ways only to track the ball, and not the players so I could achieve high FPS? I tried to make max_player_detections to 0, but I think it's not an computationally efficient way.

2) To track other sizes of soccer balls (ex. 2x larger than the Issia dataset) , would changing BALL_BBOX_SIZE be enough? Or do I also have to change other values?

It would be much appreciated if you could give me ideas of the questions above. Thanks again for this wonderful contribution!

**Comment by jac99** [comment]
Hi,

thanks for your interest in our method.
1. Unfortunately there's no an easy way to 'switch off' the player detection and use only ball detection. Our model (deep neural network) is designed to detect both the ball and players. Setting max_player_detection to 0 will not speed up processing. The network will nerveless process the entire image and produce a ball and player confidence map. max_player_detection parameter only controls maximum number of players retrieved from the player confidence map. But is very fast operation and turning it off by setting max_player+detection to 0 will no speed up things.
If you want to speed up things, I suggest:
- Use a good GPU, like nVidia RTX 3070
- Change the network design, by removing upper layers (denoted Conv4, Conv5 on the diagram) and player detection head; modify the loss function to take into account only ball-related loss and re-train the model. But I do not think it will greatly improve the performance
- Consider downsampling input images before processing

2. Changing BALL_BBOX_SIZE will not help.
Our pre-trainel model was trained on ISSIA (ball and player) and SoccerPlayerDetection_bmvc17_v1 dataset (only players). If your dataset has different characteristics (e.g. the ball is bigger) the performance will be worse (or the ball/players may not be detected at all). You have the following options:
- re-train the model, adding your dataset to the training data (you'll need a dataset with manually annotated ball position); this would required modification of the code to write the code to read your dataset etc.
- consider downsampling the input video, so the ball size matches the size of the ball in ISSIA dataset - this would speed up the processing (the smaller the images, the faster the processing) and could improve the ball detection ratio
